web/docs/deployment/deployment-methods/self-hosted.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Self-Hosted
+---
+
+If you have your own server or you rent out a server, you can self-host your Wasp apps. This deployment option has the benefit of giving you full control over your apps and its data. Also, it can be more cost-effective than using a cloud provider since you can deploy multiple apps on a single server.
+
+## What you'll need
+
+To successfully self-host your Wasp app, you need to have the following:
+- A server with a public IP address
+  - There are many cloud providers you can use to rent a server. Some popular ones are [AWS](https://aws.amazon.com/ec2/), [DigitalOcean](https://www.digitalocean.com/), [OVH](https://www.ovhcloud.com/en/vps/), and [Hetzner](https://www.hetzner.com/cloud/).
+- A domain name (we'll need this to have HTTPS)
+
+## Self-hosting steps
+
+To self-host your Wasp app, you need to follow these general steps:
+1. Get your app's code on the server. How you do this depends on your setup, we'll explore a few methods in the next section.
+2. Set up [Docker](https://docs.docker.com/engine/install/) on the server. This is needed to run the server and the database.
+3. Set up a reverse proxy to be able to use a domain name with your app.
+4. Set up the server env variables.
+
+![Self-hosting steps](/img/deploying/self-hosting.png)
+
+## Deployment methods
+
+We'll explore a few methods you can use to self-host your Wasp app. One is to use a simple setup where you manually set up the server. The other two methods will use self-hosted PaaS platforms that make it easy to deploy apps.
+
+### Simple setup
+
+In this setup, you install Docker on your server, set up the server env variable, set up a reverse proxy, and run your app. This is a very manual process, but you'll learn how everything works.
+
+<details>
+<summary>Overview of the steps</summary>
+
+1. Install Docker, Node.js and Wasp on your server.
+2. Set up a reverse proxy.
+    - We are using [Caddy](https://caddyserver.com/) in our example, but you can use any reverse proxy you like (e.g., Nginx, Apache).
+3. Get your app's code on the server with Git.
+4. Build the app with Wasp CLI.
+5. Build and run the server with Docker.
+7. Start the database with Docker.
+8. Set up Caddy to serve your client and server.
+9. Point your domain to your server's IP address.
+
+</details>
+
+Read our [full guide](https://gist.github.com/infomiho/80f3f50346566e39db56c5e57fefa1fe) on deploying Wasp apps to your VPS.
```

Comment:
Have you gone through the gists? There is a lot of content there which we would need to maintain in the docs. Keeping the specific instructions enable us to keep them more like a guide with a lot of small instructions (if we move the gists to some guides section in the docs, this will be then perfect) and not going into bigger topics and explaining every little piece. The gist is straightforward, do this, do that and you are done. I feel like that style would fall short of our docs guidelines. 

So, to conclude, I feel like until we have a guides section for all the https://github.com/wasp-lang/learning-materials/ things, I'd keep the specific deployment instructions in the gists. That then prompts me to day "what about our PaaS provider instructions"? I guess I'd move them to the gists as well.

Reply:
Ok I think I get where you are coming from.

Be it in docs under guides, or in gists, the content is the same, and so I don't think it matters much for this discussion if what we have now in gists it in gists or in docs. Would be nice for the future to have it in docs, but that is not important for this PR.

For this PR, what I think is important to notice is that these gists we have are very practical / straightforward as you said -> they don't describe how to deploy Wasp app to VPS in general, they describe how to deploy to VPS for a specific choice of tools. Which makes them so actionable / practical, but it also means, as you said, that they don't feel like the right fit for docs, unless they are marked as guides. I agree with that.

So what do we want to have? I would say that we want to have, in docs, a general guide on how to do this. I think you tried to do that with the overviews that you put into details, but I think those fall short, they rely a bit too much on you checking out the actual gist.

What if we left the gists as they are for now (they can become guides later, when we have time, but for now it doesn't matter), and we reference them as a very specific example they can check out, but we don't treat them as the main source of knowledge here in these docs -> they are just an addition. Instead, we take what you started doing in `details` and exapdn that to be a standalone piece of docs that says how to do these kinds of deployments in general. I wouldn't go overly specific in it, but I would also make it more instructive then it is now. Somewhere in the middle in between what you have right now and the gists.

Reply:
>  I wouldn't go overly specific in it

This is the main challenge in my opinion, how to communicate enough information without explaining every possible option e.g. different reverse proxies, different ways of deploying Node.js, different ways of serving the client etc.

Reply:
Coolify example:
<img width="842" alt="Screenshot 2024-12-16 at 15 31 29" src="https://github.com/user-attachments/assets/5618a569-03f5-4587-8168-e0206087e5d4" />

Simple setup example:
<img width="873" alt="Screenshot 2024-12-16 at 15 32 04" src="https://github.com/user-attachments/assets/7a03c52b-ffef-4d9d-a33e-d6a0e40ed077" />

I've expanded the instructions so they are more documentation-like and made the guide link more prominent and pointed out the specific things used (Caddy, Github Actions etc.)

Reply:
Yeah, I think this is it! You are giving me enough info to make it clera to me what is needed, but you don't go too much into the details and I can check the guide if I want for a specific example with pinned down specific choices.

In Coolify one you say "We are running the database with COolify on the same server, but you can ..." -> here  I believe you are referring to the guide below, where we made that choice? I wouldn't do that, refer to guide that way, but would instead say "You could run the database with Coolify on the same server, or ...". Max, what I would do is "You could run the database wiht Coolify on the same server (like we do in the guide below), or ...". But I think there is benefit to keeping guide a separate thing as much as we can.

And then when you link to a guide, I would have sentence in front of that that is something like "For a more detailed, step-by-step guide that demonstrates one specific way how to deploy to Coolify, check out this link", something like that. You could even mention suprt shortly choices that were made in it, if it can be said shortly.

Reply:
> Yeah, I think this is it! You are giving me enough info to make it clera to me what is needed, but you don't go too much into the details and I can check the guide if I want for a specific example with pinned down specific choices.
> 
> In Coolify one you say "We are running the database with COolify on the same server, but you can ..." -> here I believe you are referring to the guide below, where we made that choice? I wouldn't do that, refer to guide that way, but would instead say "You could run the database with Coolify on the same server, or ...". Max, what I would do is "You could run the database wiht Coolify on the same server (like we do in the guide below), or ...". But I think there is benefit to keeping guide a separate thing as much as we can.
> 
> And then when you link to a guide, I would have sentence in front of that that is something like "For a more detailed, step-by-step guide that demonstrates one specific way how to deploy to Coolify, check out this link", something like that. You could even mention suprt shortly choices that were made in it, if it can be said shortly.

I'll update the text, you are right!

Also, did you notice this in the screenshot above:
<img width="714" alt="Screenshot 2024-12-17 at 17 28 31" src="https://github.com/user-attachments/assets/34c48554-0ac0-42e9-9ff7-50feef56cab1" />

Reply:
Whoops I missed that, but that looks great, love it!

Resolve this comment when ready.